src/some-controller.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef SOME_CONTROLLER_HPP
+#define SOME_CONTROLLER_HPP
+
+#include "wrap-hwlib.hpp"
+
+class SomeController {
+  public:
+    void read_string_from_pin_in(hwlib::pin_in &in, char *buffer) {
+        while (true) {
+            char c = 0;
+
+            for (int i = 0; i < 8; i++) {
+                c <<= 1;
+                c |= in.get();
+            }
+
+            *buffer = c;
+
+            if (c == 0) {
+                break;
+            }
+
+            buffer++;
```

Comment:
This usage of the pointer to buffer seems dangerous. There is no documentation to support these choices.

Reply:
These files were added in the recent Arduino template. Will remove.